### DISCUSSION TITLE
keep coordinates after stop or disconnection problems

### DISCUSSION THREAD

**Discussion by elsolo5000-2** [winder/Universal-G-Code-Sender#1871] [Help] [answered]
sometimes during work, i have disconnections.
i must unplug replug usb to reconnect

configuration: win 7 , grbl1.1

it happens like this:
-i load a file
-i set origins
-i start the job
-and in the middle of nowhere the job stops, the machine disconnect, the spindle is running and the steppers are idle(two times the steppers where forcing this same spindle in the table, she looks less beautyfull 😭)

i already updated:
-328p bootloader
-grbl to 1.1h
-usb windows drivers
-ch340 drivers

i suspected grbl but after trying it without engine and steppers it ran 12000lines without problem
so i suspect the controller board or windows driver (who knows🤷‍♂️)
i think the board is bugging when i use too much current or the usb cable disconnect

and it got me thinked "is there a way ugs keep last coordinates for when i reconnect the board?"
because when i reconnect coordinates are reseted to some values but not the good ones.
and when i stop a job, coordinates are reseted too, why?

i know i can zero it or re-enter last coordinate but it will be a nice feature to do not have to retype or verify coordinate system

**Comment by breiler** [comment, marked as answer]
This sounds like the classical problem with electromagnetic interference. I had a lot of problems with my spindle running 220v which caused loads of electromagnetic noise, triggering limit switches. This was solved by using a properly shielded power cable to the spindle and running the limit switches on 12 volts through a relay board. I also added a shielded box for my Arduino.

This is a really common problem and there are a lot of discussions and tips to how to resolve it.

Here are some good tips n' tricks to get you started:
https://www.cnccookbook.com/cnc-electrical-noise-grounding-techniques-interference-filter/

**Reply by elsolo5000-2** [reply]
Thanks for your Reply, i will investigate into that.
And what about when i push stop on the interface, work offsets (g92) are changing? 
The thing is when i place a stock , i set coordinates offset for the job.
If for a reason i push stop, i loose or reset the coordinates. IS there a reason ? Am i doing something wrong?

**Reply by breiler** [reply]
I am not sure, I don't have this problem when using the built in "reset zero" myself.

There is some information here regarding G92 here: https://github.com/winder/Universal-G-Code-Sender/issues/125

**Reply by elsolo5000-2** [reply]
As an update, i found a filter inside a broken power drill, since i put it on the spindle the controller work well.
And for the coordinate problem it was because i used g92 instead of G10😑.
Still Thanks you for your help🤗
